.github/workflows/build_and_test.yml

```diff
@@ -1270,6 +1299,7 @@ jobs:
       - build-and-test-locally
       - check-codestyle-python
       - check-codestyle-rust
+      - check-dependencies-rust
```

Comment:
Will `conclusion ` pass if `check-codestyle-rust` gets skipped?

Reply:
No, if it's skipped then it will fail. Why are we even checking for `contains(needs.*.result, 'skipped')` though? I don't see how skipping a job should result in a CI failure. If the dependencies of a job aren't met and it doesn't run for that reason, the result should be `cancelled`, not `skipped`, right? And if a job is correctly skipped, then we shouldn't fail on it not running.

Reply:
> Why are we even checking for contains(needs.*.result, 'skipped') though? 

If all jobs are skipped, for example, the PR will still be allowed to merge, which we don’t want to happen.

We should explicitly allow a job to be skipped iff there are no changes in relevant files

Reply:
> If all jobs are skipped, for example, the PR will still be allowed to merge, which we don’t want to happen.

Why? If all jobs are skipped because their conditions say they don't need to run, then it should be okay, unless the conditions are wrong?

Either way: We can't fix this without listing all jobs in conclusion and checking explicitly for skipping, because we can't fail on `contains(needs.*.result, 'skipped')` anymore if there's a single job that's okay to skip sometimes.

Reply:
> Why? If all jobs are skipped because their conditions say they don't need to run, then it should be okay, unless the conditions are wrong?

We want them to be skipped because there are no changes in relevant files, but don't want because of some complex if-condition and a failed dependency which ends up in job being marked as skipped (and we had such example in the past).

We can do the same for the merge queue: https://github.com/neondatabase/neon/blob/361222ebbd8f8e6963a631b0e7a244d64a4e435d/.github/workflows/pre-merge-checks.yml#L127

Reply:
We have this line:

https://github.com/neondatabase/neon/blob/9f81828429ad6475b4fbb1a814240213b74bec63/.github/workflows/build_and_test.yml#L1347

If we want to do something similar to what we do for the merge queue, then we need to get rid of that line, *and* we have to add a line like the one you've linked above for every single other job that `conclusion` depends on.

Reply:
Yep, looks good!
And for a particular job, it should be `==` instead of `contains`, i.e. `needs.check-dependencies-rust.result == 'skipped'` and so on

Reply:
I've adjusted it in pre-merge-checks accordingly and added the rust job skip check as well